cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -116,8 +116,10 @@ communicate_ghosts_to_owners(MPI_Comm comm, std::span<const int> src,
     send_sizes.reserve(1);
     recv_sizes.reserve(1);
     MPI_Request sizes_request;
-    MPI_Ineighbor_alltoall(send_sizes.data(), 1, MPI_INT32_T, recv_sizes.data(),
-                           1, MPI_INT32_T, comm0, &sizes_request);
+    MPI_Ineighbor_alltoall(send_sizes.data(), 1,
+                           dolfinx::MPI::mpi_t<std::int32_t>, recv_sizes.data(),
+                           1, dolfinx::MPI::mpi_t<std::int32_t>, comm0,
```

Comment:
I'm not convinced that `dolfinx::MPI::mpi_t<std::int32_t>` adds anything over `MPI_INT32_T`. The latter is simpler.

`dolfinx::MPI::mpi_t` is helpful where we have templated over the type being communicated.

Reply:
Not the perfect interface for sure, but it allows us to stick to one type system in the code base.

We could also go for something like `dolfinx::MPI::mpi_t<sendsizes::value_type>`, which removes the possibility of a type mismatch, but increases code complexity for the caller.